raphtory/src/db/api/mutation/addition_ops.rs

```diff
@@ -278,49 +284,100 @@ impl<G: InternalAdditionOps<Error: Into<GraphError>> + StaticGraphViewOps> Addit
         props: PII,
         layer: Option<&str>,
     ) -> Result<EdgeView<G, G>, GraphError> {
+        // Start & log the transaction
+        let txn_id = self.transaction_manager().begin();
```

Comment:
Food for thought on better management of transactions


let transaction_manager = self.transaction_manager();
transaction_manager.transact(|txid,wal| {
    g.add_edge(1,2)
    wal.lod_add_edge(txn_id,...).unwrap();
})